apps/studio/src/routes/editor/LayersPanel/Tree/TreeNode.tsx

```diff
@@ -138,6 +138,16 @@ const TreeNode = observer(
             node.data.isVisible = !node.data.isVisible;
         }
 
+        function hasComponentAncestor(node: NodeApi<LayerNode>): boolean {
```

Comment:
This is a very expensive operation. With a large enough tree, the app will grind to a halt.

Reply:
Fixed! Good thing too because I added some extra performance fix thanks to this.